### PR TITLE
fix(community): hide the filter rail when there is nothing to narrow

### PR DESCRIPTION
**Pull request by andymai**
Follow-up to #3264.

`FilterRail` rendered on its open state alone, with no check that there was anything to filter. Against an empty index that means every control in it is dead — all eight categories disabled at 0, all ten technique pills disabled, all three size sliders disabled on null windows, every show-toggle at 0 — sitting in a 240px column next to "No community designs yet".

The public index is empty today (`/api/community` returns `{"items":[],"nextCursor":null,"likedIds":[]}` with a 200), so this is exactly what the live gallery shows. The old filter sheet hid all of it behind a button, so an empty gallery just showed the empty state.

## Change

The rail and the toolbar's filter toggle now require at least one loaded card. Neither the empty state nor the initial-load skeletons get a column of disabled controls beside them.

The chip row is deliberately left alone, and that matters for one case: the Mine toggle lives *inside* the rail, so an owner with nothing published would be stranded in an empty Mine view with no control to leave it. Because chips render whenever the panel is not showing, the Mine chip appears and clears it. There is a test for that exact round trip.

## Why the original tests missed it

Every unit test was fixture-backed and every browser check ran against 60 stubbed cards, so nothing exercised the one state production is actually in.

## Verification

- 20,719 unit tests pass; typecheck clean; 0 lint errors
- New coverage: rail and toggle absent over an empty index, rail returns once cards load, and the empty-Mine escape hatch

---
## Summary by cubic
Hide the filter rail and toolbar toggle when there’s nothing to narrow. Also prevent the mobile filter view from reopening by itself when filters return.

- **Bug Fixes**
  - Gate panel visibility inside `useFilterPanel`; clear the mobile state when nothing is available while leaving the desktop rail preference intact.
  - Hide the filter toggle in the toolbar unless at least one card is loaded.
  - Keep chips visible so users can clear the “Mine” filter when the rail is hidden.
  - Add tests for empty index, rail returning after items load, the empty-Mine escape, and the mobile view not auto-reopening.

<sup>Written for commit fe0bdee74bd7c0c7abcbdd138884e318076928d6. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/andymai/gridfinity-layout-tool/pull/3266?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: src/features/community/components/CommunityGalleryTab/CommunityGalleryTab.tsx [hunk 1]
```diff
@@ -200,7 +200,13 @@ export function CommunityGalleryTab({
...
   // fresh scroll container. The offset was banked by handleTogglePanel while
   // the old one was still on screen.
-  const mobileFiltersOpen = isMobile && filterPanel.open;
+  // Nothing loaded means nothing to narrow, and every control in the panel
+  // would be a dead one: all counts zero, every category and technique
+  // disabled, every size axis with an empty window. The empty state stands on
+  // its own instead.
+  const filtersAvailable = activeItems.length > 0;
+  const panelOpen = filterPanel.open && filtersAvailable;
+  const mobileFiltersOpen = isMobile && panelOpen;
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/community/components/CommunityGalleryTab/CommunityGalleryTab.tsx:206]: Good catch, real bug. Fixed in fe0bdee.

`useFilterPanel` now takes `filtersAvailable` and owns the gating, so the two states diverge the way they should: dropping to nothing-to-narrow clears the mobile state outright, while the rail preference is left untouched and the rail returns on its own once cards load. That matches the split already there for the breakpoint reset — a rail reappearing is recoverable, a full-screen takeover nobody asked for is not.

Two tests added, including your exact scenario (open the mobile view, switch to an empty Mine index, clear it, view stays closed).
